spec/models/spree/gateway/stripe_sca_spec.rb

```diff
@@ -71,6 +63,77 @@
     end
   end
 
+  describe "#void", :vcr, :stripe_version do
+    # This is the first account retrieved using Stripe::Account.list
+    let(:stripe_test_account) { "acct_1OhZ9lQQeZbvfZRJ" }
+
+    before do
+      # Inject our test stripe account
+      stripe_account = create(:stripe_account, stripe_user_id: stripe_test_account)
+      allow(StripeAccount).to receive(:find_by).and_return(stripe_account)
+    end
+
+    context "with a confirmed payment" do
+      it "refunds the payment" do
+        # Link the payment intent to our test stripe account, and automatically confirm and capture
+        # the payment.
+        payment_intent = Stripe::PaymentIntent.create(
+          {
+            amount: 1000, # given in AUD cents
+            currency: 'aud', # AUD to match order currency
+            payment_method: 'pm_card_mastercard',
+            payment_method_types: ['card'],
+            capture_method: 'automatic',
+            confirm: true,
+          },
+          stripe_account: stripe_test_account
+        )
+
+        payment = create(
+          :payment,
+          order:,
+          amount: order.total,
+          payment_method: subject,
+          source: credit_card,
+          response_code: payment_intent.id
+        )
+
+        response = subject.void(payment_intent.id, nil, {})
+
+        expect(response.success?).to eq true
+      end
+    end
+
+    context "with a voidable payment" do
+      it "void the payment" do
+        # Link the payment intent to our test stripe account
+        payment_intent = Stripe::PaymentIntent.create(
+          {
+            amount: 1000, # given in AUD cents
+            currency: 'aud', # AUD to match order currency
+            payment_method: 'pm_card_mastercard',
+            payment_method_types: ['card'],
+            capture_method: 'manual'
+          },
+          stripe_account: stripe_test_account
+        )
+
+        payment = create(
+          :payment,
+          order:,
+          amount: order.total,
+          payment_method: subject,
+          source: credit_card,
+          response_code: payment_intent.id
+        )
```

Comment:
This setup is almost the same as the the above, so it would be great to DRY it a bit. If `payment` and `payment_intent` are moved to `let`s, the `payment` can be shared at a higher level.